defaulting.go

```diff
@@ -0,0 +1,29 @@
+package ruleengine
+
+import "context"
+
+type DefaultingRule struct {
+	Conditions []func() bool
+	Defaulting func()
+}
+
+func ExecuteDefaulting(ctx context.Context, rules []DefaultingRule) {
+	for _, rule := range rules {
+		if rule.Conditions != nil {
+			skip := false
+
+			for _, condition := range rule.Conditions {
+				if !condition() {
+					skip = true
```

Comment:
This might be a matter of taste, but I'd personally find it easier to read/comprehend when using `goto SKIP` instead of having need to follow boolean flag logic for skipping. WDYT?

Something similar to this: https://play.golang.org/p/jzGKCU-fwV_n

Reply:
Tried it and the tests fail. I think skip might be executed regardless in some cases. I would not spend more time here now and keep it as it is.

Reply:
Ok. I'm fine with that.